playbooks/pulsys_user.yml

```diff
@@ -48,6 +48,11 @@
         mode: '0600'
         backup: true
 
+    - name: set fact for hosts matched in the first play
+      ansible.builtin.set_fact:
+        first_play_had_hosts: true
+      when: ansible_play_hosts_all is search("princeton.edu")
```

Comment:
Can you do `ansible_play_hosts_all|length>0` ?

Reply:
Ah, good idea! I was thinking about whether `[ ]` would have length of greater than 0, and then the lightbulb came on. We don't need to set a condition at all! We can just set the variable. If we get `no hosts matched` the entire play gets skipped, so the variable won't get set.